### PR TITLE
OAK-9123: fix immense term error + performance regression on multi term full text queries

### PR DESCRIPTION
**Pull request by fabriziofortino**

### CODE REVIEW

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java [hunk 1]
```diff
@@ -479,7 +479,10 @@ private static QueryBuilder tokenToQuery(String text, String fieldName, PlanResu
             // TODO: do we need this if all the analyzed fields are queried?
             ret = in.should(matchQuery(fieldName, text));
```
Comment (nit0906) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java]: shouldn't we have an AND operator here too ?

**Review by nit0906 — COMMENTED** [review]
LGTM. Just a small comment.

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java (hunk 1 above)
Comment (fabriziofortino) [reply oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java]: correct, let me add a test to capture this case

**Review by nit0906 — APPROVED** [review]
lgtm

**Pull request closed without merging**